### PR TITLE
docs(collaboration): update collaboration

### PR DESCRIPTION
**Pull request by ianforster-sage**
### Proposed behaviour
This is an update to the COLLABORATION.md file to better describe our processes today. 

### Current behaviour
Currently the COLLABORATION.md file is inaccurate and incomplete.

### Additional context
We will continue to add to the guidelines with styleguides for:
- Test specs
- Documentation
- JavaScript
Plus a Code of Conduct will be added.

### Testing instructions
Documentation only, no testing required.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,65 +1,146 @@
...
-* Ensure descriptive `.yml` file is added to the `./change/next` directory.
+## Pull Request Guidelines
+Before submitting a pull request, check the [issue tracker](https://github.com/sage/carbon/issues) to see if the feature or bug has already been discussed. If it has, check with us before beginning work on it to avoid duplicated effort. If no issue has been raised, please raise one first before beginning work. You can then link back to this when submitting a pull request to address it.
+
+Pull requests can be used in conjunction with two workflows:
+### The **Feature Branch** workflow 
+If you are a Carbon maintainer you should develop your feature in a branch of the Carbon repository. To merge into `master` you must first open a pull request with your feature branch as the source and `master` as the destination.
+
+### The **Forking** workflow 
+If you are a third party contributor you should push your feature to your own public repository, this will be the pull request source and the destination will typically be the Carbon repo `master` branch.
```
Comment (chrisbarber86) [inline CONTRIBUTING.md]:
```suggestion
If you are a third party contributor, you should fork the carbon repository and make your changes there. When ready you should then raise a pull request to the `master` branch of carbon.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,65 +1,146 @@
...
+
+Whichever workflow you use when you create the pull request you should include a detailed description of the feature you are adding and ensure you have included information on how to setup and QA your new feature or bug-fix.
+
+Your branch should meets the following criteria:
+
+1. Release notes are included.
```
Comment (jamime) [inline CONTRIBUTING.md]: Strictly speaking we dont write release notes, but it's worth mentioning git message format here.

Also, theres often confusion around working on a feature branch and the message. e.g. if you break something on a feature branch there is no need to do a `fix` commit. Instead they should squash the fix with the commit that broke it.

If they don't do this they will end up with a patch release for a bug that was never released.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,65 +1,154 @@
...
+1. footers other than `BREAKING CHANGE: <description> `may be provided and follow a convention similar to [git trailer format](https://git-scm.com/docs/git-interpret-trailers).
+
+**Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.  
+[[Source]](https://github.com/angular/angular/blob/2089727db988c8d3336e06c10fc75049565305ad/CONTRIBUTING.md#commit)
+
+NB. When working on your feature branch, if you introduce a bug on that branch only and then fix it in a later commit you should squash this commit into the commit where you introduced the bug. There is no need to create a `fix` commit for a bug that didn't make it into production.
+
+We use [`commitlint`](https://github.com/conventional-changelog/commitlint) and [`husky`](https://github.com/typicode/husky) to check our commits against the `conventionalcommits` guidelines.
+- [`husky`](https://github.com/typicode/husky) manages the git commit hooks defined in `package.json`
+- we write a hook that runs the commit message via [`commitlint`](https://github.com/conventional-changelog/commitlint)
```
Comment (jamime) [inline CONTRIBUTING.md:126]:
```suggestion
- we have a hook that runs the commit message via [`commitlint`](https://github.com/conventional-changelog/commitlint)
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,65 +1,154 @@
...
+- we use the [`@commitlint/config-conventional`](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) config
+- [`husky`](https://github.com/typicode/husky) aborts the commit if the commit message is in the wrong format
+
+We use [`commitizen`](https://github.com/commitizen/cz-cli) to help write commit messages
+- we use the [`cz-conventional-changelog`](https://github.com/commitizen/cz-conventional-changelog) config
+- you can choose to write commit messages with the regular `git` CLI or in their IDE still
```
Comment (jamime) [inline CONTRIBUTING.md:132]:
```suggestion
- you can choose to write commit messages with the regular `git` CLI or in your IDE still
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,65 +1,154 @@
...
+- you can launch [`commitizen`](https://github.com/commitizen/cz-cli) with `npm run commit`
+- If a commit message is discarded by `husky` e.g. if we run the tests in the `pre-commit` hook and they fail; you can use `npm run commit -- --retry` to retry with the same commit message.
```
Comment (jamime) [inline CONTRIBUTING.md:134]:
```suggestion
- If you used `commitizen` and your commit is discarded by `husky` e.g. if we run the tests in the `pre-commit` hook and they fail; you can use `npm run commit -- --retry` to retry with the same commit message. 
```

**Pull request merged**